lib/dynamoid/criteria/chain.rb

```diff
@@ -598,6 +598,11 @@ def raw_pages_via_scan
         end
       end
 
+      def validate_scan_conditions
```

Comment:
code climate was complaining about cognitive complexity so I moved this portion of the method out